### ISSUE TITLE
Disks don't get mounted when creating zos-vm on a zeroos-host

### ISSUE THREAD

**Issue by john-kheir** [threefoldtech/qemu#3]
## Scenario
1- Create a disk without a filesystem
```
zdb = zos_client.primitives.create_zerodb(name='myzdb', path='/mnt/zdbs/sda',
                                                  mode='user', sync=False, admin='mypassword')
zdb.namespaces.add(name='mynamespace', size=50, password='namespacepassword', public=True)
zdb.deploy()
disk = zos_client.primitives.create_disk('mydisk', zdb, mountpoint='/mnt', filesystem='ext4', size=50)
disk.deploy()
```

2- Create a zos-vm using this disk on any zos-host

3- Check if the disk is mounted 

## Actual Result:
Disk is not mounted. 
```
In [8]: cl.bash('df -h').get()
Out[8]: 
STATE: 0 SUCCESS
STDOUT:
Filesystem                Size      Used Available Use% Mounted on
tmpfs                   512.0M    326.1M    185.9M  64% /
devtmpfs                  7.8G         0      7.8G   0% /dev
cgroup_root               7.8G         0      7.8G   0% /sys/fs/cgroup
overlay                 512.0M    326.1M    185.9M  64% /mnt/containers/1
tmpfs                   512.0M    326.1M    185.9M  64% /mnt/containers/1/root/jumpscale/cfg
tmpfs                   512.0M    326.1M    185.9M  64% /mnt/containers/1/root/.ssh
tmpfs                   512.0M    326.1M    185.9M  64% /mnt/containers/1/tmp/redis.sock
tmpfs                   512.0M    326.1M    185.9M  64% /mnt/containers/1/opt/code/local/stdorg/config
tmpfs                   512.0M    326.1M    185.9M  64% /mnt/containers/1/opt/var/data/zrobot/zrobot_data
tmpfs                   512.0M    326.1M    185.9M  64% /mnt/containers/1/coreX

STDERR:

DATA:

```

## Notes:
```
In [10]: cl.bash('lsblk -o NAME,TYPE,LABEL').get()
Out[10]: 
STATE: 0 SUCCESS
STDOUT:
NAME   TYPE LABEL
vda    disk 
`-vda1 part sp_zos-cache

STDERR:

DATA:


In [11]: cl.bash('btrfs filesystem show').get()
Out[11]: 
STATE: 0 SUCCESS
STDOUT:
Label: 'sp_zos-cache'  uuid: 9712af40-4f58-4f89-bcf6-eaeefef2d3c1
	Total devices 1 FS bytes used 112.00KiB
	devid    1 size 50.00GiB used 2.02GiB path /dev/vda1


STDERR:

DATA:
```
The disk can't be even mounted manually
```

In [12]: cl.bash('mkdir -p /mnt/test && mount /dev/vda1 /mnt/test').get()
Out[12]: 
STATE: 1255 ERROR
STDOUT:

STDERR:
mount: mounting /dev/vda1 on /mnt/test failed: Invalid argument

DATA:


```

**Comment by john-kheir** [comment]
```
In [13]: cl.bash('dmesg | tail ').get()
Out[13]: 
STATE: 0 SUCCESS
STDOUT:
[  898.024355] print_req_error: I/O error, dev vda, sector 75968
[  898.024360] BTRFS error (device vda1): bdev /dev/vda1 errs: wr 8, rd 0, flush 0, corrupt 0, gen 0
[  898.024407] BTRFS: error (device vda1) in btrfs_commit_transaction:2253: errno=-5 IO failure (Error while writing out transaction)
[  898.024411] BTRFS warning (device vda1): Skipping commit of aborted transaction.
[  898.024417] BTRFS: error (device vda1) in cleanup_transaction:1873: errno=-5 IO failure
[  898.024419] BTRFS info (device vda1): delayed_refs has NO entry
[  898.024443] BTRFS warning (device vda1): failed to create the UUID tree: -5
[  898.024460] BTRFS error (device vda1): commit super ret -30
[  898.024673] BTRFS error (device vda1): cleaner transaction attach returned -30
[  898.041672] BTRFS error (device vda1): open_ctree failed

STDERR:

DATA:
```

**Comment by john-kheir** [comment]
Verified

**Comment by maxux** [comment]
Fixed in 54bdf9a0b921eb2a029e0f10d03f14a265e24ecb
